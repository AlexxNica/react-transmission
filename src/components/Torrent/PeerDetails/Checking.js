import React from 'react';
import { FormattedMessage, injectIntl } from 'react-intl';

function Checking({ torrent, intl }) {
  const { recheckProgress } = torrent;

  return (
    <FormattedMessage
      id='torrent.full.checking'
      defaultMessage={`Verifying local data ({recheckProgress} tested)`}
      values={{
        recheckProgress: intl.formatNumber(recheckProgress, {style: 'percent'}),
      }}
    />
  );
}

export default injectIntl(Checking);
