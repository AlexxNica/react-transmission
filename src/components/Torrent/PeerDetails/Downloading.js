import React from 'react';
import { FormattedMessage } from 'react-intl';

import { formatUL, formatDL } from 'util/formatters';

function DownloadingFromPeersAndWebSeeds({ torrent }) {
  const { peersSendingToUs, peersConnected, webseedsSendingToUs, rateUpload, rateDownload } = torrent;

  return (
    <FormattedMessage
      id='torrent.full.downloading.frompeersandwebseeds'
      defaultMessage={`Downloading from {peersSendingToUs, number} of {peersConnected, number} {peersConnected, plural, one {peer} other {peers}} and {webseedsSendingToUs, number} {webseedsSendingToUs, plural, one {web seed} other {web seeds}} - {formattedRateDownload} {formattedRateUpload}`}
      values={{
        peersSendingToUs,
        peersConnected,
        webseedsSendingToUs,
        formattedRateUpload: formatUL(rateUpload),
        formattedRateDownload: formatDL(rateDownload),
      }}
    />
  );
}

function DownloadingFromWebSeeds({ torrent }) {
  const { webseedsSendingToUs, rateUpload, rateDownload } = torrent;

  return (
    <FormattedMessage
      id='torrent.full.downloading.fromwebseeds'
      defaultMessage={`Downloading from {webseedsSendingToUs, number} {webseedsSendingToUs, plural, one {web seed} other {web seeds}} - {formattedRateDownload} {formattedRateUpload}`}
      values={{
        webseedsSendingToUs,
        formattedRateUpload: formatUL(rateUpload),
        formattedRateDownload: formatDL(rateDownload),
      }}
    />
  );
}

function DownloadingFromPeers({ torrent }) {
  const { peersSendingToUs, peersConnected, rateUpload, rateDownload } = torrent;

  return (
    <FormattedMessage
      id='torrent.full.downloading.frompeers'
      defaultMessage={`Downloading from {peersSendingToUs, number} of {peersConnected, number} {peersConnected, plural, one {peer} other {peers}} - {formattedRateDownload} {formattedRateUpload}`}
      values={{
        peersSendingToUs,
        peersConnected,
        formattedRateUpload: formatUL(rateUpload),
        formattedRateDownload: formatDL(rateDownload),
      }}
    />
  );
}

export default function({ torrent }) {
  const { peersConnected, webseedsSendingToUs } = torrent;

  if (webseedsSendingToUs && peersConnected) {
    return <DownloadingFromPeersAndWebSeeds torrent={torrent} />
  }

  if (webseedsSendingToUs) {
    return <DownloadingFromWebSeeds torrent={torrent} />
  }

  return <DownloadingFromPeers torrent={torrent} />
}
