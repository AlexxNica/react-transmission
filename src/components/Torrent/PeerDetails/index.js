import React from 'react';

function PeerDetails({ torrent }) {
  return (
    <p>
      <strong>{torrent.totalSize}</strong>
    </p>
  );
}

export default PeerDetails;
