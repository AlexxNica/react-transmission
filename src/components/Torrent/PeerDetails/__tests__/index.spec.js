import React from 'react';
import createComponentWithIntl from 'test/util/createComponentWithIntl';

import PeerDetails from '../';

test('PeerDetails has errors', () => {
  const component = createComponentWithIntl(
    <PeerDetails torrent={{hasErrors: true}} />
  );

  expect(component.toJSON()).toMatchSnapshot();
});

const torrent = {
  recheckProgress: 1,
  peersGettingFromUs: 5,
  peersSendingToUs: 5,
  peersConnected: 5,
  webseedsSendingToUs: 5,
  rateUpload: 1E5,
  rateDownload: 1E7,
};

test('PeerDetails downloading', () => {
  const component = createComponentWithIntl(
    <PeerDetails torrent={{...torrent, isDownloading: true}} />
  );

  expect(component.toJSON()).toMatchSnapshot();
});

test('PeerDetails seeding', () => {
  const component = createComponentWithIntl(
    <PeerDetails torrent={{...torrent, isSeeding: true}} />
  );

  expect(component.toJSON()).toMatchSnapshot();
});

test('PeerDetails checking', () => {
  const component = createComponentWithIntl(
    <PeerDetails torrent={{...torrent, isChecking: true}} />
  );

  expect(component.toJSON()).toMatchSnapshot();
});

test('PeerDetails status', () => {
  const component = createComponentWithIntl(
    <PeerDetails torrent={{...torrent}} />
  );

  expect(component.toJSON()).toMatchSnapshot();
});
