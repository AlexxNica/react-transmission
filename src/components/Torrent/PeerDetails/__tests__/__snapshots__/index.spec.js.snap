exports[`test PeerDetails checking 1`] = `
<span>
  Verifying local data (100% tested)
</span>
`;

exports[`test PeerDetails downloading 1`] = `
<span>
  Downloading from 5 of 5 peers and 5 web seeds - ↓ 10.00 MB/s ↑ 100 kB/s
</span>
`;

exports[`test PeerDetails has errors 1`] = `null`;

exports[`test PeerDetails seeding 1`] = `
<span>
  Seeding to 5 of 5 connected peers - ↑ 100 kB/s
</span>
`;

exports[`test PeerDetails status 1`] = `
<span>
  Unknown
</span>
`;
